packages/hdwallet-provider/src/constructor/getOptions.ts

```diff
@@ -102,12 +101,9 @@ const fromArguments = (
 const matchesNewInputOptions = (
   args: ConstructorArguments
 ): args is [Constructor.InputOptions] => {
-  // new-style means exactly one argument
-  if (args.length !== 1) {
-    return false;
-  }
-
   const [options] = args;
+  // new-style means exactly one argument and an object
+  if (args.length !== 1 || typeof options !== "object") return false;
```

Comment:
This (and the message at the end) is the only real change, right?  The rest is all `prettier`?

Reply:
yes correct

Reply:
I mean, it's not really a bug - it is just an improvement of the validation and clearer messaging. If you feel strongly that this is a fix then we can go ahead and change it.

Reply:
Well this isn't an internal improvement; it has actual external effects!  So it's either an enhancement or a bug fix; I'd call it a bug fix, as it's causing something that ought to have worked to actually work.